### PR TITLE
Use final on classes where possible

### PR DESCRIPTION
**Pull request by ChrisKujawa**
## Description

We decided in the team meeting to use final as default on classes, to encourage developers to favor composition over inheritance. Furthermore before we try to extend classes we should think about whether we really need that or not.

Interesting reads to that topic:

 * https://medium.com/@lukleDev/how-effective-java-may-have-influenced-the-design-of-kotlin-part-2-89844d62ddf3
 * https://blog.frankel.ch/open-your-classes-and-methods-in-kotlin/
 * https://matthiasnoback.nl/2018/09/final-classes-by-default-why/
 * https://ocramius.github.io/blog/when-to-declare-classes-final/
 * http://www.javapractices.com/topic/TopicAction.do?Id=23
 * http://whiley.org/2011/12/06/final-should-be-default-for-classes-in-java/

## Related issues

I run the following script and on compile errors I removed the finals again.

```
#!/bin/bash

for file in $(find $1 -name *.java) 
do
  sed -i 's/public class/public final class/g' $file
done
```

So no deep refactoring.

I ignored the following modules:

 * bpmn-model
 * exporters/elasticsearch-exporter
 * dist
 * test

## Pull Request Checklist

- [X] All commit messages match our [commit message guidelines](https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md#commit-message-guidelines)
- [X] The submitting code follows our [code style](https://github.com/zeebe-io/zeebe/wiki/Code-Style)
- [X] If submitting code, please run `mvn clean install -DskipTests` locally before committing

### CODE REVIEW

**Comment by npepinpe** [comment]
I guess merging the other PR caused tons of conflicts :sweat_smile: I'll wait until you rebase to review, but we can do this tomorrow (the hack day seems like a good day to do this kind of major change).

**Comment by Zelldon** [comment]
I updated the branch

**Comment by Zelldon** [comment]
I fixed all reported errors as well. Could you please have a look @npepinpe

File: gateway/src/main/java/io/zeebe/gateway/impl/broker/BrokerClientImpl.java [hunk 1]
```diff
@@ -167,6 +167,7 @@ public void close() {
       try {
         actorScheduler.stop().get(15, TimeUnit.SECONDS);
       } catch (final InterruptedException | ExecutionException | TimeoutException e) {
+        Thread.currentThread().interrupt();
```
Comment (npepinpe) [inline gateway/src/main/java/io/zeebe/gateway/impl/broker/BrokerClientImpl.java]: This will mark the thread has interrupted on `TimeoutException`, or?

**Review by npepinpe — CHANGES_REQUESTED** [review]
:tada: 

Thanks a lot for doing this. Only one tiny thing

File: gateway/src/main/java/io/zeebe/gateway/impl/broker/BrokerClientImpl.java (hunk 1 above)
Comment (Zelldon) [reply gateway/src/main/java/io/zeebe/gateway/impl/broker/BrokerClientImpl.java]: yes sorry will change this

**Review by npepinpe — APPROVED** [review]
:+1: 

Just merge when you've updated the PR

**Pull request merged**